From a6b0f0a1faa6991424d247e3b0323bc638445aa8 Mon Sep 17 00:00:00 2001
From: Artur Gebicz <a.gebicz@metrological.com>
Date: Fri, 31 Jul 2020 11:03:28 +0200
Subject: [PATCH] [DisplayInfo] Use DisplayStatus for obtaining HDR status

---
 DisplayInfo/Nexus/PlatformImplementation.cpp | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/DisplayInfo/Nexus/PlatformImplementation.cpp b/DisplayInfo/Nexus/PlatformImplementation.cpp
index 82cba93..e932b26 100644
--- a/DisplayInfo/Nexus/PlatformImplementation.cpp
+++ b/DisplayInfo/Nexus/PlatformImplementation.cpp
@@ -46,7 +46,7 @@ public:
        , _height(0)
        , _verticalFreq(0)
        , _connected(false)
-       , _hdcpprotection(HDCP_Unencrypted)
+       , _hdcpprotection(HDCPProtectionType::HDCP_Unencrypted)
        , _type(HDR_OFF)
        , _totalGpuRam(0)
        , _audioPassthrough(false)
@@ -390,9 +390,15 @@ private:
                 NxClient_DisplaySettings displaySettings;
                 NxClient_GetDisplaySettings(&displaySettings);
 
+                NxClient_DisplayStatus status;
+                NEXUS_Error rcStatus = NxClient_GetDisplayStatus(&status);
+                if(rcStatus != NEXUS_SUCCESS){
+                    TRACE_L1(_T("Failed to get display status with rc=%d", rcStatus));
+                }
+
 #ifdef NEXUS_HDR_SUPPORTED
                 // Read HDR status
-                switch (displaySettings.hdmiPreferences.dynamicRangeMode) {
+                switch (status.hdmi.dynamicRangeMode) {
                 case NEXUS_VideoDynamicRangeMode_eHdr10:
                     hdr = HDR_10;
                     break;
-- 
2.17.1

