diff --git a/DIALServer/DIALServer.cpp b/DIALServer/DIALServer.cpp
index c3de995..aa1683c 100644
--- a/DIALServer/DIALServer.cpp
+++ b/DIALServer/DIALServer.cpp
@@ -277,7 +277,7 @@ namespace Plugin {
         _config.FromString(service->ConfigLine());
 
         // get an interface with a public IP address, then we will have a proper MAC address..
-        Core::NodeId selectedNode = Plugin::Config::IPV4UnicastNode(_config.Interface.Value());
+        Core::NodeId selectedNode = Core::NodeId(_config.Interface.Value().c_str());
 
         if (selectedNode.IsValid() == false) {
             // Oops no way we can operate...
