From 3dd1287c1a75a3a3d64e3e40860dae9267452899 Mon Sep 17 00:00:00 2001
From: Bram Oosterhuis <mail@bybram.com>
Date: Tue, 6 Oct 2020 20:36:50 +0200
Subject: [PATCH 2/5] DIAL: network interface workaround

---
 DIALServer/DIALServer.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/DIALServer/DIALServer.cpp b/DIALServer/DIALServer.cpp
index 2c90af1..cf37c49 100644
--- a/DIALServer/DIALServer.cpp
+++ b/DIALServer/DIALServer.cpp
@@ -278,7 +278,7 @@ namespace Plugin {
         _config.FromString(service->ConfigLine());
 
         // get an interface with a public IP address, then we will have a proper MAC address..
-        Core::NodeId selectedNode = Plugin::Config::IPV4UnicastNode(_config.Interface.Value());
+        Core::NodeId selectedNode = Core::NodeId(_config.Interface.Value().c_str());
 
         if (selectedNode.IsValid() == false) {
             // Oops no way we can operate...
-- 
2.25.1

