From bfe2271b270354222ae85250af22ce1bd7448015 Mon Sep 17 00:00:00 2001
From: Bram Oosterhuis <mail@bybram.com>
Date: Thu, 8 Oct 2020 10:59:05 +0200
Subject: [PATCH 3/5] DIAL: Add flag to switch deprecated API

---
 DIALServer/DIALServer.h | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/DIALServer/DIALServer.h b/DIALServer/DIALServer.h
index 989bea9..d6515fd 100644
--- a/DIALServer/DIALServer.h
+++ b/DIALServer/DIALServer.h
@@ -106,6 +106,7 @@ namespace Plugin {
                 , Interface()
                 , WebServer()
                 , SwitchBoard()
+                , DeprecatedAPI(true)
             {
                 Add(_T("interface"), &Interface);
                 Add(_T("name"), &Name);
@@ -119,6 +120,7 @@ namespace Plugin {
                 Add(_T("upc"), &UPC);
                 Add(_T("webserver"), &WebServer);
                 Add(_T("switchboard"), &SwitchBoard);
+                Add(_T("deprecatedapi"), &DeprecatedAPI);
                 Add(_T("apps"), &Apps);
             }
             ~Config()
@@ -138,6 +140,7 @@ namespace Plugin {
             Core::JSON::String Interface;
             Core::JSON::String WebServer;
             Core::JSON::String SwitchBoard;
+            Core::JSON::Boolean DeprecatedAPI;
             Core::JSON::ArrayType<App> Apps;
         };
 
@@ -1005,6 +1008,7 @@ namespace Plugin {
             , _deviceInfo(Core::ProxyType<Web::TextBody>::Create())
             , _sink(this)
             , _appInfo()
+            , _deprecatedAPI(false)
         {
         }
 #ifdef __WINDOWS__
@@ -1067,6 +1071,8 @@ namespace Plugin {
         void event_stop(const string& application, const string& parameters);
         void event_hide(const string& application);
 
+        inline const bool DeprecatedAPI() const { return _deprecatedAPI;}
+
     private:
         Core::CriticalSection _adminLock;
         uint32_t _skipURL;
@@ -1079,6 +1085,7 @@ namespace Plugin {
         Core::ProxyType<Web::TextBody> _deviceInfo;
         Core::Sink<Notification> _sink;
         std::map<const string, AppInformation> _appInfo;
+        bool _deprecatedAPI;
     };
 }
 }
-- 
2.25.1

