--- a/gst/gstprotection.h	2019-10-26 13:10:20.974858321 +0200
+++ b/gst/gstprotection.h	2019-10-26 13:11:14.790194670 +0200
@@ -34,6 +34,24 @@
  */
 #define GST_PROTECTION_SYSTEM_ID_CAPS_FIELD "protection-system"
 
+/**
+ * GST_PROTECTION_UNSPECIFIED_SYSTEM_ID:
+ *
+ * The protection system value of the unspecified UUID.
+ * In some cases the system protection ID is not present in the contents or in their
+ * metadata, as encrypted WebM.
+ * This define is used to set the value of the "system_id" field in GstProtectionEvent,
+ * with this value, the application will use an external information to choose which
+ * protection system to use.
+ *
+ * Example: The matroskademux uses this value in the case of encrypted WebM,
+ * the application will choose the appropriate protection system based on the information
+ * received through EME API.
+ *
+ * Since: 1.16
+ */
+#define GST_PROTECTION_UNSPECIFIED_SYSTEM_ID "unspecified-system-id"
+
 typedef struct _GstProtectionMeta GstProtectionMeta;
 /**
  * GstProtectionMeta:
