From 6341e33d17106d809c7b7aef41442db914a8d9f2 Mon Sep 17 00:00:00 2001
From: Bram Oosterhuis <mail@bybram.com>
Date: Thu, 5 Nov 2020 15:52:09 +0100
Subject: [PATCH] Load the gstreamerplayer at startup

---
 .../implementation/src/LinuxDeviceLayerCommon.cpp | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/thunder/linux-device-layer/implementation/src/LinuxDeviceLayerCommon.cpp b/thunder/linux-device-layer/implementation/src/LinuxDeviceLayerCommon.cpp
index f5a65d3a1..428f85869 100644
--- a/thunder/linux-device-layer/implementation/src/LinuxDeviceLayerCommon.cpp
+++ b/thunder/linux-device-layer/implementation/src/LinuxDeviceLayerCommon.cpp
@@ -12,6 +12,12 @@
 #include <thunder/IIgnitionCommunication.h>
 #include <AmazonBackendPlatform.h>
 
+#include <IPlayer.h>
+
+extern "C" {
+Amazon::IPlayer* PlayerFactory(); // found in libamazon-backend.so
+}
+
 namespace {
 const std::string INVALID_ROOT_DIR = "";
 
@@ -80,7 +86,16 @@ namespace Ignite {
     {
         if(_platform != nullptr) {
             _platform->Initialize(config);
+
+            // FIXME: Hacky I know, but the first player loaded is has no stable playback. 
+            // Load it here the first time when the backend is loaded. This needs to be 
+            // fixed properly or ditch the gstreamerplayer. 
+            PlayerFactory()->Initialise();
+            sleep(1);  // give gstreamer some time to initialise... 
+            PlayerFactory()->Deinitialise();
+
             return AV_OK;
+
         } else {
             return -1;
         }
-- 
2.25.1

