diff --git a/Source/WebCore/platform/graphics/gstreamer/eme/WebKitCommonEncryptionDecryptorGStreamer.cpp b/Source/WebCore/platform/graphics/gstreamer/eme/WebKitCommonEncryptionDecryptorGStreamer.cpp
index 77106e8..d81c0b7 100644
--- a/Source/WebCore/platform/graphics/gstreamer/eme/WebKitCommonEncryptionDecryptorGStreamer.cpp
+++ b/Source/WebCore/platform/graphics/gstreamer/eme/WebKitCommonEncryptionDecryptorGStreamer.cpp
@@ -247,6 +247,11 @@ static GstFlowReturn webkitMediaCommonEncryptionDecryptTransformInPlace(GstBaseT
                 return GST_FLOW_NOT_SUPPORTED;
             }
         }
+        if (priv->m_isFlushing) {
+             GST_INFO_OBJECT(self, "key received but drop this frame");
+             return GST_FLOW_FLUSHING;
+        }
+
         GST_DEBUG_OBJECT(self, "key received, continuing");
     }
 
@@ -488,10 +493,12 @@ static gboolean webkitMediaCommonEncryptionDecryptSinkEventHandler(GstBaseTransf
     case GST_EVENT_FLUSH_START: {
         {
             LockHolder locker(priv->m_mutex);
-            ASSERT(!priv->m_isFlushing);
-            priv->m_isFlushing = true;
-            GST_DEBUG_OBJECT(self, "flushing");
-            priv->m_condition.notifyOne();
+            if (!priv->m_keyReceived) {
+                ASSERT(!priv->m_isFlushing);
+                priv->m_isFlushing = true;
+                GST_DEBUG_OBJECT(self, "flushing");
+                priv->m_condition.notifyOne();
+            }
         }
         result = GST_BASE_TRANSFORM_CLASS(parent_class)->sink_event(trans, event);
         break;
@@ -499,9 +506,11 @@ static gboolean webkitMediaCommonEncryptionDecryptSinkEventHandler(GstBaseTransf
     case GST_EVENT_FLUSH_STOP: {
         {
             LockHolder locker(priv->m_mutex);
-            ASSERT(priv->m_isFlushing);
-            priv->m_isFlushing = false;
-            GST_DEBUG_OBJECT(self, "flushing done");
+            if (!priv->m_keyReceived) {
+                ASSERT(priv->m_isFlushing);
+                priv->m_isFlushing = false;
+                GST_DEBUG_OBJECT(self, "flushing done");
+            }
         }
         result = GST_BASE_TRANSFORM_CLASS(parent_class)->sink_event(trans, event);
         break;
