diff --git a/Source/WebKit/Shared/glib/ProcessExecutablePathGLib.cpp b/Source/WebKit/Shared/glib/ProcessExecutablePathGLib.cpp
index 89f135070838..427b3157b436 100644
--- a/Source/WebKit/Shared/glib/ProcessExecutablePathGLib.cpp
+++ b/Source/WebKit/Shared/glib/ProcessExecutablePathGLib.cpp
@@ -46,14 +46,15 @@ static String getExecutablePath()
 
 static String findWebKitProcess(const char* processName)
 {
-#if ENABLE(DEVELOPER_MODE)
     static const char* execDirectory = g_getenv("WEBKIT_EXEC_PATH");
     if (execDirectory) {
         String processPath = FileSystem::pathByAppendingComponent(FileSystem::stringFromFileSystemRepresentation(execDirectory), processName);
         if (FileSystem::fileExists(processPath))
+            printf("WEBKIT_EXEC_PATH is set, alternative path for WebKit processes: %s\n", execDirectory);
             return processPath;
     }
 
+#if ENABLE(DEVELOPER_MODE)
     static String executablePath = getExecutablePath();
     if (!executablePath.isNull()) {
         String processPath = FileSystem::pathByAppendingComponent(executablePath, processName);
