diff --git a/src/third_party/starboard/wpe/shared/window/window_get_size.cc b/src/third_party/starboard/wpe/shared/window/window_get_size.cc
index 80cd396..4829e21 100644
--- a/src/third_party/starboard/wpe/shared/window/window_get_size.cc
+++ b/src/third_party/starboard/wpe/shared/window/window_get_size.cc
@@ -17,6 +17,9 @@
 
 #include "third_party/starboard/wpe/shared/window/window_internal.h"
 
+#define MAX_SUPPORTED_RESOLUTION_WIDTH 3840
+#define MAX_SUPPORTED_RESOLUTION_HEIGHT 2160
+
 bool SbWindowGetSize(SbWindow window, SbWindowSize* size) {
   if (!SbWindowIsValid(window)) {
     SB_DLOG(ERROR) << __FUNCTION__ << ": Invalid window.";
@@ -25,6 +28,8 @@ bool SbWindowGetSize(SbWindow window, SbWindowSize* size) {
 
   size->width = static_cast<SbWindowPrivate*>(window)->window_->Width();
   size->height = static_cast<SbWindowPrivate*>(window)->window_->Height();
-  size->video_pixel_ratio = 1.0f;
+  size->video_pixel_ratio = std::max(
+    static_cast<float>(MAX_SUPPORTED_RESOLUTION_WIDTH / size->width),
+    static_cast<float>(MAX_SUPPORTED_RESOLUTION_HEIGHT / size->height));
   return true;
 }
