diff --git a/kpn-metrological/platform/ruby/amp-kpn-metrological/libs/Hawaii/HawaiiBindings/src/hawaii/bindings/PosixBindings.cpp b/kpn-metrological/platform/ruby/amp-kpn-metrological/libs/Hawaii/HawaiiBindings/src/hawaii/bindings/PosixBindings.cpp
index 52793ff..5071a67 100644
--- a/kpn-metrological/platform/ruby/amp-kpn-metrological/libs/Hawaii/HawaiiBindings/src/hawaii/bindings/PosixBindings.cpp
+++ b/kpn-metrological/platform/ruby/amp-kpn-metrological/libs/Hawaii/HawaiiBindings/src/hawaii/bindings/PosixBindings.cpp
@@ -153,6 +153,8 @@ extern "C"
 
 	bool initBindings()
 	{
+        
+#if 0 // FIXME: OZGUR: Should be re-enabled for deployment
 		/*
 		 * IMPORTANT! This code must not be altered when porting. It is to
 		 * prevent the Amazon Video Player from being used in a process running
@@ -162,7 +164,7 @@ extern "C"
 		uid_t userId = getuid();
 		core::Log::write( core::Log::kInfo, "Initialising Bindings with User Id: %d\n", userId );
 		AMP_VALIDATE( userId != 0, ::error::kError_InvalidUserPermissions );
-
+#endif
 		if ( !posixBindings )
 		{
 			posixBindings = new hawaii::bindings::PosixBindings();
