From ed57180dd52ce335d18b48fd3a37a66817f72045 Mon Sep 17 00:00:00 2001
From: "k.plata" <k.plata@metrological.com>
Date: Fri, 17 Sep 2021 14:29:48 +0200
Subject: [PATCH] Stash seek event until we reach paused state

---
 AmazonBackendPlayer.cpp | 28 +++++++++++++++++++++-------
 1 file changed, 21 insertions(+), 7 deletions(-)

diff --git a/AmazonBackendPlayer.cpp b/AmazonBackendPlayer.cpp
index da85dce..4fa4cf5 100644
--- a/AmazonBackendPlayer.cpp
+++ b/AmazonBackendPlayer.cpp
@@ -684,7 +684,8 @@ namespace Amazon {
                 , _contextThread(nullptr)
                 , _callback(callback)
                 , _drmLibrary(amazonLibrary)
-                , _ocdmSession(nullptr) {
+                , _ocdmSession(nullptr)
+                , _seekPending(false) {
 
                 // initialize gst
                 GstServer::Instance()->Connect();
@@ -816,12 +817,16 @@ namespace Amazon {
                 }
 
                 if (time || seeking) {
-
-                    gst_element_seek(_pipeline, 1.0, GST_FORMAT_TIME,
-                                          static_cast<GstSeekFlags>( GST_SEEK_FLAG_FLUSH | GST_SEEK_FLAG_ACCURATE),
-                                          GST_SEEK_TYPE_SET, static_cast<gint64>(time), GST_SEEK_TYPE_SET,
-                                          _video ? static_cast<gint64>(_video->Duration() * SECONDS_TO_NANOSECONDS)
-                                                 : static_cast<gint64>(0.0));
+                    GstState currentState;
+                    gst_element_get_state(_pipeline, &currentState, nullptr, 250 * GST_NSECOND);
+                    if (currentState >= GST_STATE_PAUSED )
+                    {
+                        gst_element_seek_simple(_pipeline, GST_FORMAT_TIME, static_cast<GstSeekFlags>( GST_SEEK_FLAG_FLUSH | GST_SEEK_FLAG_ACCURATE), time);
+
+                    } else {
+		               _seekPending = true;
+		               _pendingSeekTime = time;
+		           }
                 }
                 ChangeState(GST_STATE_PLAYING);
 
@@ -1005,6 +1010,12 @@ namespace Amazon {
                         if (g_strcmp0(GST_MESSAGE_SRC_NAME(message), "media_pipeline") == 0 /*&& newState == GST_STATE_PLAYING*/) {
                             GST_DEBUG_BIN_TO_DOT_FILE_WITH_TS(GST_BIN (self->_pipeline), GST_DEBUG_GRAPH_SHOW_ALL, "pipeline");
                         }
+                        if(newState == GST_STATE_PAUSED && currentState == GST_STATE_READY && self->_seekPending && g_strcmp0(GST_MESSAGE_SRC_NAME(message), "media_pipeline") == 0) {
+                            gst_element_seek_simple(self->_pipeline, GST_FORMAT_TIME, static_cast<GstSeekFlags>(GST_SEEK_FLAG_ACCURATE), self->_pendingSeekTime);
+                            self->_seekPending = false;
+                            self->_pendingSeekTime = 0;
+                        }
+
                         self->UpdateState();
                         break;
                     case GST_MESSAGE_BUFFERING:
@@ -1425,6 +1436,9 @@ namespace Amazon {
             WPEFramework::Core::Library _drmLibrary;
             OCDMSessionFunction _ocdmSessionFunction;
             OpenCDMSession *_ocdmSession;
+
+            bool _seekPending;
+            uint64_t _pendingSeekTime;
         };
 
         enum class State {
-- 
2.25.1

