diff -Naur ThunderClientLibraries.orig/Source/compositorclient/Client.h ThunderClientLibraries/Source/compositorclient/Client.h
--- ThunderClientLibraries.orig/Source/compositorclient/Client.h	2022-05-19 13:47:18.000000000 +0000
+++ ThunderClientLibraries/Source/compositorclient/Client.h	2022-12-05 16:59:37.989207342 +0000
@@ -190,6 +190,7 @@
         virtual ISurface* Create(const std::string& name, const uint32_t width, const uint32_t height) = 0; //initial position on screen is fullscreen,x and y therefore implicit and 0
         virtual int Process(const uint32_t data) = 0;
         virtual int FileDescriptor() const = 0;
+	virtual ISurface* SurfaceByName(const std::string& name) = 0;
     };
 } // Compositor
 } // WPEFramework
