diff --git a/partner/dpi/gstreamer/FileSystem.cpp b/partner/dpi/gstreamer/FileSystem.cpp
index eaaf7e0d..8a89f0ea 100644
--- a/partner/dpi/gstreamer/FileSystem.cpp
+++ b/partner/dpi/gstreamer/FileSystem.cpp
@@ -410,7 +410,7 @@ void FileSystem::HandleVideoProfiles()
 
 #ifdef USE_DISPLAY_INFO
     enable50 &= (mOutputConnected && (mOutputHeight >= 2160) && (mHdcpVersion >= HDCP_2_2));
-    enable51 &= (mOutputConnected && (mOutputHeight >= 2160) && (mOutputFrameRate >= 5000) && (mHdcpVersion >= HDCP_2_2));
+    enable51 &= (mOutputConnected && (mOutputHeight >= 2160) && (mHdcpVersion >= HDCP_2_2));
     enableHDR = !((mHDRType == DISPLAYINFO_HDR_OFF) || (mHDRType == DISPLAYINFO_HDR_UNKNOWN));
     IDEBUG("4K VIDEO PROFILE LEVELS:%s%s%s", enable50?" 5.0":"", enable51?" 5.1":"", !enable50&&!enable51?" NONE":"");
 #endif // USE_DISPLAY_INFO
