From b935c3bd08c798fa1df09e8ee3ac2519accc0176 Mon Sep 17 00:00:00 2001
From: pwielders <developper@wielders.net>
Date: Tue, 25 Dec 2018 18:25:35 +0100
Subject: [PATCH] The pc file is named accordingly to the sources.

---
 cmake/FindLibVirtualKeyboard.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cmake/FindLibVirtualKeyboard.cmake b/cmake/FindLibVirtualKeyboard.cmake
index c4f48e1..d8f4a12 100644
--- a/cmake/FindLibVirtualKeyboard.cmake
+++ b/cmake/FindLibVirtualKeyboard.cmake
@@ -7,7 +7,7 @@
 # Copyright (C) 2016 Metrological.
 #
 find_package(PkgConfig)
-pkg_check_modules(LIBVIRTUAL_KEYBOARD REQUIRED virtualkeyboard)
+pkg_check_modules(LIBVIRTUAL_KEYBOARD REQUIRED WPEFrameworkVirtualInput)
 
 include(FindPackageHandleStandardArgs)
 FIND_PACKAGE_HANDLE_STANDARD_ARGS(LIBVIRTUAL_KEYBOARD DEFAULT_MSG
-- 
2.11.0

