diff --git a/Netflix.cpp b/Netflix.cpp
index 68e41de..e04fb81 100644
--- a/Netflix.cpp
+++ b/Netflix.cpp
@@ -108,19 +108,18 @@ namespace Plugin {
             _notification.Release();
         }
         
-        if (_netflix->Release() != Core::ERROR_DESTRUCTION_SUCCEEDED) {
+        _netflix->Release();
 
-            ASSERT(_connectionId != 0);
+        ASSERT(_connectionId != 0);
 
-            TRACE_L1("Netflix Plugin is not properly destructed. %d", _connectionId);
+        TRACE_L1("Netflix Plugin is not properly destructed. %d", _connectionId);
 
-            RPC::IRemoteConnection* connection(_service->RemoteConnection(_connectionId));
+        RPC::IRemoteConnection* connection(_service->RemoteConnection(_connectionId));
 
-            // The process can disappear in the meantime...
-            if (connection != nullptr) {
-                connection->Terminate();
-                connection->Release();
-            }
+        // The process can disappear in the meantime...
+        if (connection != nullptr) {
+            connection->Terminate();
+            connection->Release();
         }
 
         // Deinitialize what we initialized..
