diff --git a/cmake/project.cmake.in b/cmake/project.cmake.in
index e3fac76a..c26f0523 100644
--- a/cmake/project.cmake.in
+++ b/cmake/project.cmake.in
@@ -36,13 +36,6 @@ endif()
 
 set(NAMESPACE "@NAMESPACE@")
 
-option(EXPORT_SYMBOLS "Export to all symbols" OFF)
-
-if (NOT EXPORT_SYMBOLS)
-    set(CMAKE_CXX_VISIBILITY_PRESET hidden)
-    set(CMAKE_VISIBILITY_INLINES_HIDDEN 1)
-endif()
-
 set(MODULE_BASE_DIR "${CMAKE_CURRENT_LIST_DIR}")
 
 list(APPEND CMAKE_MODULE_PATH
