diff --git a/WebKitBrowser/WebKitImplementation.cpp b/WebKitBrowser/WebKitImplementation.cpp
index aa33d2d..1ee5970 100644
--- a/WebKitBrowser/WebKitImplementation.cpp
+++ b/WebKitBrowser/WebKitImplementation.cpp
@@ -336,6 +336,8 @@ static GSourceFuncs _handlerIntervention =
                 , Width(1280)
                 , Height(720)
                 , PTSOffset(0)
+                , VideoPTSOffset(0)
+                , AudioPTSOffset(0)
                 , ScaleFactor(1.0)
             {
                 Add(_T("useragent"), &UserAgent);
@@ -372,6 +374,8 @@ static GSourceFuncs _handlerIntervention =
                 Add(_T("height"), &Height);
                 Add(_T("ptsoffset"), &PTSOffset);
                 Add(_T("scalefactor"), &ScaleFactor);
+                Add(_T("videoptsoffset"), &VideoPTSOffset);
+                Add(_T("audioptsoffset"), &AudioPTSOffset);
             }
             ~Config()
             {
@@ -411,6 +415,8 @@ static GSourceFuncs _handlerIntervention =
             Core::JSON::DecUInt16 Width;
             Core::JSON::DecUInt16 Height;
             Core::JSON::DecSInt16 PTSOffset;
+            Core::JSON::DecUInt16 VideoPTSOffset;
+            Core::JSON::DecUInt16 AudioPTSOffset;
             Core::JSON::DecUInt16 ScaleFactor;
         };
 
@@ -804,6 +810,18 @@ static GSourceFuncs _handlerIntervention =
                 Core::SystemInfo::SetEnvironment(_T("GST_VIRTUAL_DISP_HEIGHT"), height, !environmentOverride);
             }
 
+            // VideoPTSOffset
+            if (_config.VideoPTSOffset.IsSet() == true) {
+                string videoptsoffset(Core::NumberType<uint16_t>(_config.VideoPTSOffset.Value()).Text());
+                Core::SystemInfo::SetEnvironment(_T("GST_VIDEO_PTS_OFFSET_MS"), videoptsoffset, !environmentOverride);
+            }
+
+            // AudioPTSOffset
+            if (_config.AudioPTSOffset.IsSet() == true) {
+                string audioptsoffset(Core::NumberType<uint16_t>(_config.AudioPTSOffset.Value()).Text());
+                Core::SystemInfo::SetEnvironment(_T("GST_AUDIO_PTS_OFFSET_MS"), audioptsoffset, !environmentOverride);
+            }
+
             // Oke, so we are good to go.. Release....
             Core::Thread::Run();
 
