From a5a3c28474932c2b572c70aca5687632a4cf5ebb Mon Sep 17 00:00:00 2001
From: Artur Gebicz <a.gebicz@metrological.com>
Date: Wed, 5 Aug 2020 14:48:11 +0200
Subject: [PATCH 1/2] [interfaces] Extend DisplayInfo::HDRType with HGL(HDR)

---
 Source/interfaces/IDisplayInfo.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Source/interfaces/IDisplayInfo.h b/Source/interfaces/IDisplayInfo.h
index 62a2a07c..5ba563a4 100644
--- a/Source/interfaces/IDisplayInfo.h
+++ b/Source/interfaces/IDisplayInfo.h
@@ -42,6 +42,7 @@ namespace Exchange {
             HDR_OFF,
             HDR_10,
             HDR_10PLUS,
+            HDR_HLG,
             HDR_DOLBYVISION,
             HDR_TECHNICOLOR
         };
-- 
2.17.1

