From fe8cfa13abe33df62b9836815dba851014564107 Mon Sep 17 00:00:00 2001
From: Artur Gebicz <a.gebicz@metrological.com>
Date: Wed, 26 Aug 2020 16:05:37 +0200
Subject: [PATCH] [interfaces] Add AtmosLock functionality to Dolby::IOutput

---
 Source/interfaces/IDolby.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/Source/interfaces/IDolby.h b/Source/interfaces/IDolby.h
index 81076236..e6fb3606 100644
--- a/Source/interfaces/IDolby.h
+++ b/Source/interfaces/IDolby.h
@@ -40,8 +40,9 @@ namespace Exchange {
             };
 
             virtual void Mode(const Dolby::IOutput::Type value) = 0;
-            
             virtual Dolby::IOutput::Type Mode() const = 0;
+            virtual bool AtmosLock(uint32_t& error /* @out */) const = 0;
+            virtual uint32_t AtmosLock(bool lock) = 0;
         };
     }
 }
-- 
2.17.1

