From 269e1e3fc5556e394787ed6543ac7c2c7762336f Mon Sep 17 00:00:00 2001
From: Bram Oosterhuis <mail@bybram.com>
Date: Mon, 16 Nov 2020 17:55:12 +0100
Subject: [PATCH] displayinfo: add prefix to error codes

---
 Source/displayinfo/include/displayinfo.h | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/Source/displayinfo/include/displayinfo.h b/Source/displayinfo/include/displayinfo.h
index e89be66..8de31a1 100644
--- a/Source/displayinfo/include/displayinfo.h
+++ b/Source/displayinfo/include/displayinfo.h
@@ -56,9 +56,9 @@ typedef enum displayinfo_hdcp_protection_type {
 } displayinfo_hdcp_protection_t;
 
 typedef enum displayinfo_error_type {
-    ERROR_NONE = 0,
-    ERROR_UNKNOWN = 1,
-    ERROR_INVALID_INSTANCE = 2,
+    DISPLAYINFO_ERROR_NONE = 0,
+    DISPLAYINFO_ERROR_UNKNOWN = 1,
+    DISPLAYINFO_ERROR_INVALID_INSTANCE = 2,
 } displayinfo_error_t;
 
 /**
-- 
2.25.1

