From 964f243d4e4fea6a3c3b0910f1d343d5b6c4681a Mon Sep 17 00:00:00 2001
From: Bram Oosterhuis <mail@bybram.com>
Date: Tue, 17 Nov 2020 21:44:25 +0100
Subject: [PATCH] displayinfo: return actual memory values

---
 Source/displayinfo/DisplayInfo.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Source/displayinfo/DisplayInfo.cpp b/Source/displayinfo/DisplayInfo.cpp
index 84f62a0..a081868 100644
--- a/Source/displayinfo/DisplayInfo.cpp
+++ b/Source/displayinfo/DisplayInfo.cpp
@@ -434,12 +434,12 @@ public:
     uint64_t TotalGpuRam() const 
     {
         uint64_t memory(0);
-        return _graphicsProperties != nullptr ? _graphicsProperties->TotalGpuRam(memory) : memory;
+        return (_graphicsProperties != nullptr ? (_graphicsProperties->TotalGpuRam(memory) == Core::ERROR_NONE ? memory : 0) : 0);
     }
     uint64_t FreeGpuRam() const 
     {
         uint64_t memory(0);
-        return _graphicsProperties != nullptr ? _graphicsProperties->FreeGpuRam(memory) : memory;
+        return (_graphicsProperties != nullptr ? (_graphicsProperties->FreeGpuRam(memory) == Core::ERROR_NONE ? memory : 0) : 0);
     }
 
 private:
-- 
2.25.1

