diff --git a/Tools/JsonGenerator/JsonGenerator.py b/Tools/JsonGenerator/JsonGenerator.py
index 1dad99c1..5c7113c1 100755
--- a/Tools/JsonGenerator/JsonGenerator.py
+++ b/Tools/JsonGenerator/JsonGenerator.py
@@ -1336,7 +1336,7 @@ def EmitRpcCode(root, emit, header_file, source_file, data_emitted):
     emit.Line()
     emit.Line("#pragma once")
     emit.Line()
-    emit.Line("#include \"Module.h\"")
+    emit.Line("#include \"../Module.h\"")
     if data_emitted:
         emit.Line("#include \"%s_%s.h\"" % (DATA_NAMESPACE, header_file))
     emit.Line("#include <%s%s>" % (CPP_IF_PATH, source_file))
